File: Common/Board/Pawn.cpp
#include "Pawn.h"

Pawn::Pawn()
{
    Clear();
}

Pawn::Pawn(unsigned short xp, unsigned short yp, unsigned short deadp, unsigned short ponsp)
{
    x = xp;
    y = yp;
    dead = deadp;
    pons = ponsp;
}

void Pawn::Clear()
{
    dead = 1;
    pons = 0;
}

File: Java/CheckersClinet/src/pl/boards/Pawn.java
package pl.boards;

public class Pawn
{	

	public Pawn()
	{
	    Clear();
	}

	public Pawn(final Pawn val)
	{
	    x = val.x;
	    y = val.y;
	    dead = val.dead;
	    pons = val.pons;
	}
	
	public Pawn(int xp, int yp, boolean deadp, boolean ponsp)
	{
	    x = xp;
	    y = yp;
	    dead = deadp;
	    pons = ponsp;
	}
	
	public void Clear()
	{
	    dead = true;
	    pons = false;
	}

    public int x = 4;
    public int y = 4;
    public boolean dead = true;
    public boolean pons = true;
}
